src/lib.rs

```diff
@@ -6,14 +6,20 @@
 //!
 //! # Usage
 //! ```
-//! use obd2::{commands::Obd2DataRetrieval, device::Elm327, Obd2};
+//! use obd2::{commands::Obd2DataRetrieval, device::{Elm327, FTDIDevice}, Obd2};
 //!
 //! fn main() -> Result<(), obd2::Error> {
-//!     let mut device = Obd2::<Elm327>::default();
+//!     let mut device = Obd2::<Elm327::<FTDIDevice>>::new(Elm327::new(FTDIDevice::new()?)?)?;
```

Comment:
Will this work?
```suggestion
//!     let mut device = Obd2::new(Elm327::<FTDIDevice>::new(FTDIDevice::new()?)?)?;
```

I think it will, and it removes some duplication, but I can't check because of the missing code. If it does probably apply it below as well.

Reply:
resolved